nntrainer/tensor/manager.cpp

```diff
@@ -376,7 +376,9 @@ std::vector<Weight *> Manager::requestWeights(
    *  and therefore, if we remove the calcDerivative order, then tests fails.
    */
 
-  TensorLifespan var_ls = TensorLifespan::MAX_LIFESPAN;
+  TensorLifespan var_ls = swap_mode == "inference"
+                            ? TensorLifespan::FORWARD_INFER_LIFESPAN
+                            : TensorLifespan::MAX_LIFESPAN;
   TensorLifespan grad_ls = TensorLifespan::BACKWARD_FUNC_LIFESPAN;
```

Comment:
This is a pure question (it may be irrelevant to this PR, though).
Do we need `grad_ls` for the inference swap mode? Is it related to the loss thing?